src/os/user/iterate_plan9_test.go

```diff
@@ -0,0 +1,124 @@
+package user
+
+import (
+	"reflect"
+	"testing"
+)
+
+var wantUsers = []*User{
+	{
+		Uid:      "-1",
+		Gid:      "-1",
+		Username: "adm",
+		Name:     "adm",
+		HomeDir:  "/usr/adm",
+	},
+	{
+		Uid:      "0",
+		Gid:      "0",
+		Username: "none",
+		Name:     "none",
+		HomeDir:  "/usr/none",
+	},
+	{
+		Uid:      "1",
+		Gid:      "1",
+		Username: "tor",
+		Name:     "tor",
+		HomeDir:  "/usr/tor",
+	},
+	{
+		Uid:      "2",
+		Gid:      "2",
+		Username: "glenda",
+		Name:     "glenda",
+		HomeDir:  "/usr/glenda",
+	},
+	{
+		Uid:      "9999",
+		Gid:      "9999",
+		Username: "noworld",
+		Name:     "noworld",
+		HomeDir:  "/usr/noworld",
+	},
+	{
+		Uid:      "10000",
+		Gid:      "10000",
+		Username: "sys",
+		Name:     "sys",
+		HomeDir:  "/usr/sys",
+	},
+	{
+		Uid:      "10001",
+		Gid:      "10001",
+		Username: "upas",
+		Name:     "upas",
+		HomeDir:  "/usr/upas",
+	},
+	{
+		Uid:      "10002",
+		Gid:      "10002",
+		Username: "bootes",
+		Name:     "bootes",
+		HomeDir:  "/usr/bootes",
+	},
+	{
+		Uid:      "test",
+		Gid:      "test",
+		Username: "test",
+		Name:     "test",
+		HomeDir:  "/usr/test",
+	},
+}
+
+var wantGroups = []*Group{
+	{Name: "adm", Gid: "-1"},
+	{Name: "none", Gid: "0"},
+	{Name: "tor", Gid: "1"},
+	{Name: "glenda", Gid: "2"},
+	{Name: "noworld", Gid: "9999"},
+	{Name: "sys", Gid: "10000"},
+	{Name: "upas", Gid: "10001"},
+	{Name: "bootes", Gid: "10002"},
+	{Name: "test", Gid: "test"},
+}
+
+const testUsersFile = "./testdata/plan9/users.txt"
+
+func TestIterateUsers(t *testing.T) {
+	usersFile = testUsersFile
+
+	gotUsers := make([]*User, 0, len(wantUsers))
+
+	err := iterateUsers(func(user *User) error {
+		gotUsers = append(gotUsers, user)
+		return nil
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(wantUsers, gotUsers) {
+		t.Errorf("could not parse users correctly")
+	}
+}
```

Comment:
- print users when this happens
- it was probably parsed correctly, but the result was invalid.